src/v2/views/MonsterCollectionOverlay/MonsterCollectionContent.tsx

```diff
@@ -298,91 +294,33 @@ export function MonsterCollectionContent({
-                  amount={systemRewards[currentIndex!].crystal + "%"}
-                  title={
-                    <>
-                      Crystal
-                      <br />
-                      Grinding
-                    </>
-                  }
-                  isUpgrade={selectedIndex! >= currentIndex!}
-                  updatedAmount={systemRewards[selectedIndex!].crystal + "%"}
-                >
-                  <img src={crystalImg} height={48} />
-                </Item>
-                <Item
-                  key="arena"
-                  amount={systemRewards[currentIndex!].arena + "%"}
-                  title={
-                    <>
-                      Arena
-                      <br />
-                      Reward
-                    </>
-                  }
-                  isUpgrade={selectedIndex! >= currentIndex!}
-                  updatedAmount={systemRewards[selectedIndex!].arena + "%"}
-                >
-                  <img src={ncgImg} height={48} />
-                </Item>
-                <Item
-                  key="stage"
-                  amount={systemRewards[currentIndex!].stage + "% DC"}
-                  title="Stage AP"
-                  isUpgrade={selectedIndex! >= currentIndex!}
-                  updatedAmount={systemRewards[selectedIndex!].stage + "%"}
-                >
-                  <img src={apImg} height={48} />
-                </Item>
-              </ItemGroup>
-            ) : (
-              <ItemGroup key="system" title="System Rewards">
-                <Item
-                  key="crystal"
-                  amount={systemRewards[currentIndex!].crystal + "%"}
-                  title={
-                    <>
-                      Crystal
-                      <br />
-                      Grinding
-                    </>
-                  }
-                >
-                  <img src={crystalImg} height={48} />
-                </Item>
-                <Item
-                  key="arena"
-                  amount={systemRewards[currentIndex!].arena + "%"}
-                  title={
-                    <>
-                      Arena
-                      <br />
-                      Reward
-                    </>
-                  }
-                >
-                  <img src={ncgImg} height={48} />
-                </Item>
-                <Item
-                  key="stage"
-                  amount={systemRewards[currentIndex!].stage + "% DC"}
-                  title="Stage AP"
-                >
-                  <img src={apImg} height={48} />
-                </Item>
-              </ItemGroup>
-            )}
+            <ItemGroup key="system" title="System Rewards">
+              {systemRewards.map((item) => {
+                const amount = item.amount[currentIndex ?? 0];
+                const updatedAmount = item.amount[selectedIndex ?? 0];
+                const sysRewardSuffix = item.name === "stage" ? "% DC" : "%";
```

Comment:
```suggestion
                const sysRewardSuffix = item.name === "stage" ? "% DC" : "%";
                const amount = item.amount[currentIndex ?? 0] + sysRewardSuffix;
                const updatedAmount = item.amount[selectedIndex ?? 0] + sysRewardSuffix;
```

This way, we don't need to concatenate the suffix every time we display them.